lib/render/SliceRenderer.cpp

```diff
@@ -0,0 +1,700 @@
+    tf->makeLut(_cacheParams.tf_lut);
+    _cacheParams.tf_minMax = tf->getMinMaxMapValue();
+
+    if (_colorMapTextureID != 0) 
+        glDeleteTextures(1, &_colorMapTextureID);
+    
+    glGenTextures(1, &_colorMapTextureID);
+    glActiveTexture(GL_TEXTURE0);
+    glBindTexture(GL_TEXTURE_1D, _colorMapTextureID);
+    glTexParameteri(GL_TEXTURE_1D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
+    glTexParameteri(GL_TEXTURE_1D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+    glTexParameteri(GL_TEXTURE_1D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+    glTexParameteri(GL_TEXTURE_1D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+    glTexImage1D(
+        GL_TEXTURE_1D, 
+        0, 
+        GL_RGBA8, 
+        _colorMapSize,
+        0, 
+        GL_RGBA, 
+        GL_FLOAT, 
+        &_cacheParams.tf_lut[0]
+    );
+}
+   
+int SliceRenderer::_resetBoxCache() { 
+    SliceParams* p = dynamic_cast<SliceParams*>(GetActiveParams());
+    assert(p);
+    p->GetBox()->GetExtents(_cacheParams.boxMin, _cacheParams.boxMax);
+
+    int rc = _dataMgr->GetVariableExtents(
+        _cacheParams.ts,
+        _cacheParams.varName,
+        _cacheParams.refinementLevel,
+        _cacheParams.domainMin,
+        _cacheParams.domainMax
+    );
+    if (rc < 0) {
+        SetErrMsg(
+            "Unable to determine domain extents for %s", 
+            _cacheParams.varName.c_str()
+        );
+        return rc;
+    }
+    _setVertexPositions();
+    _resetTextureCoordinates();
+    return rc;
+}
+
+void SliceRenderer::_resetTextureCoordinates() {
+    float texMinX, texMinY, texMaxX, texMaxY;
+    
+    std::vector<double> boxMin = _cacheParams.boxMin;
+    std::vector<double> boxMax = _cacheParams.boxMax;
+    std::vector<double> domainMin = _cacheParams.domainMin;
+    std::vector<double> domainMax = _cacheParams.domainMax;
+
+    int xAxis, yAxis;
+    int orientation = _cacheParams.orientation;
+    if (orientation == XY) {
+        xAxis = X;
+        yAxis = Y;
+    }
+    else if (orientation == XZ) {
+        xAxis = X;
+        yAxis = Z;
+    }
+    else {   // (orientation = YZ)
+        xAxis = Y;
+        yAxis = Z;
+    }
+
+    texMinX = ( boxMin[xAxis] - domainMin[xAxis] ) / 
+              ( domainMax[xAxis] - domainMin[xAxis] );
+    texMaxX = ( boxMax[xAxis] - domainMin[xAxis] ) / 
+              ( domainMax[xAxis] - domainMin[xAxis] );
+    texMinY = ( boxMin[yAxis] - domainMin[yAxis] ) / 
+              ( domainMax[yAxis] - domainMin[yAxis] );
+    texMaxY = ( boxMax[yAxis] - domainMin[yAxis] ) / 
+              ( domainMax[yAxis] - domainMin[yAxis] );
+
+    _texCoords.clear();
+    _texCoords = {
+        texMinX, texMinY,
+        texMaxX, texMinY,
+        texMinX, texMaxY,
+        texMaxX, texMinY,
+        texMaxX, texMaxY,
+        texMinX, texMaxY
+    };
+    
+    glBindBuffer(GL_ARRAY_BUFFER, _texCoordVBO);
+    glBufferSubData(
+        GL_ARRAY_BUFFER, 
+        0, 
+        sizeof(float)*sizeof(_texCoords),
```

Comment:
You previously declared _texCoordVBO as GL_STATIC_DRAW and you are not using sizeof correctly here